zipkin-server/src/main/java/zipkin2/server/internal/ZipkinGrpcCollector.java

```diff
@@ -67,7 +69,11 @@
       try {
         CompletableFutureCallback result = new CompletableFutureCallback();
         List<Span> spans = SpanBytesDecoder.PROTO3.decodeList(bytes.nioBuffer());
-        collector.accept(spans, result);
+
+        ServiceRequestContext.mapCurrent(
```

Comment:
maybe copy same comment from collector here about rationale and blocking

contextaware for tracing and blocking because we dont yet have a way to tell if call.enqueue is actually non blocking